### ISSUE TITLE
Potential desync sources

### ISSUE THREAD

**Issue by mheguy** [Valansch/RedMew#459]
tl;dr at bottom

Premises:
1) That there is only one source of the desync. This is not a safe assumption.
2) That the desync is caused by the code itself. This is a safe assumption as [others running our code have experienced desyncs](https://discordapp.com/channels/432567222481846283/432570869039104012/516057198435237898).
3) That the desyncs are not caused by the map presets themselves. There have been reports from 3 different maps, one of them being old and known stable.

Timeline: first desync mention in the discord log is [Valansch, The Fedora Avalanche: nice desync](https://discordapp.com/channels/432567222481846283/432570869039104012/515292759330914305) at 2359 UTC Nov 25.

Two maps were active at that time: Grid since 1807 UTC Nov 20 and Turkey since 1500 UTC Nov 22.
We have had desync reports for both, so the source is from the start of grid or earlier.
The first commit after 1807 UTC Nov 20 is f9c40be920ea4583fcb52ef210a8f79db891d123, so that can be established as the later limit of the desync error.

The other limit should likely be before the start of the map previous standard map, which would be Maltease Crossing 1648 UTC Nov 15. Assuming the map maker was behind on their code, that puts the lower limit around 1648 UTC Nov 14: ebc5b58320afa3ed962751dd82d2003d386a9765 

[This](https://github.com/Refactorio/RedMew/compare/ebc5b58320afa3ed962751dd82d2003d386a9765..f9c40be920ea4583fcb52ef210a8f79db891d123) is the diff between those 2 commits.

208 changed files with 52,531 additions and 54,902 deletions.

It's a lot of changes, but only a few are in files that are always loaded and had actual content changes.

`chat_triggers` has a significant addition: mentions
`custom_commands` had major overhauls: logging was added, walkabout was split out, jail was split out, admin_chat was overhauled, reporting was overhauled
`fish_market` had messages split out, and the saviour plane hardcoded in, had checks for offline players added
`free_item_logging` had an overhaul
`player_create` was split out of control, only changes were to add the random message on first join (L22) and surface the info window (L27)
`reactor_meltdown` had a global table entry commented out because there's no reference to it anywhere else in the code (L112)
`report` had an overhaul
`train_saviour` changes to L12
`walkabout` split from custom_commands and overhauled
`info` had an overhaul
`popup` player print (subsequently fixed to use game.print_player) and logging added
`list_utils` added `table.get_random` and `table.get_random_weighted`
`utils` `print_admins` overhauled, `log_command` added, `find_entities_by_last_user` changed but is not called by any code

tl;dr: chat_triggers, custom_commands, fish_market, free_item_logging, player_create, reactor_meltdown, report, train_saviour, walkabout, info, popup, list_utils, and utils are all likely candidated.

**Comment by mheguy** [comment]
Turns out maltease cross was using a *very* old version of the scenario, so the lower limit is now the release of the winding road 3 days earlier. Commit that winding road is on: f1d4bf739c6e1842d6eace577f215ec3b20af1e1

Diff between previous lower limit and now one:
https://github.com/Refactorio/RedMew/compare/f1d4bf739c6e1842d6eace577f215ec3b20af1e1...ebc5b58

Thanksfully, no major changes in it. A little shifting in info and config, report removes an unused variable and builders adds 2 functions that only tetris has touched

**Comment by linaori** [comment]
One of the first Diggy runs already had a mass DC once, I'm not sure if it happened again afterwards, but it desynced everyone, though after re-joining, it didn't happen anymore afaik.

**Comment by linaori** [comment]
Is there any code that's loaded, listens to events and such, but doesn't really add anything? I got told that bot.lua might be a potential source of desyncs, but that's removed now.

**Comment by SimonFlapse** [comment]
Bot.lua was removed after all the latest maps was released. So it could be

**Comment by grilledham** [comment]
It's highly unlikely that bot.lua was causing desyncs. bot.lua was removed because we only needed it for prime's server.

**Comment by mheguy** [comment]
bot.lua hadn't changed since 09/23 and all it did was print to the server console

**Comment by mheguy** [comment]
Going to close this since we haven't had a desync in a couple days. Can always re-open if needed.

**Comment by mheguy** [comment]
Another desync popped up 22 hours into gear grind map, though it may have been due to the player. Will re-open this for the time being and see if more comes of it.
